internal/cmd/stencil/stencil.go

```diff
@@ -244,29 +244,67 @@ func (c *Command) Upgrade(ctx context.Context, skipRender bool) error {
 	return c.runWithModules(ctx, mods)
 }
 
-// Run fetches dependencies of the root modules and builds the layered filesystem,
-// after that GenerateFiles is called to actually walk the filesystem and render
-// the templates. This step also does minimal post-processing of the dependencies
-// manifests
-func (c *Command) Run(ctx context.Context) error {
-	c.log.Info("Fetching dependencies")
-	mods, err := c.resolveModules(ctx, false)
+// validateStencilVersion ensures that the running Stencil version is
+// compatible with the given Stencil modules.
+func (c *Command) validateStencilVersion(mods []*modules.Module, stencilVersion string) error {
+	// devel is when you're running in tests, it will always fail semver parsing.
+	if stencilVersion == "devel" {
+		return nil
+	}
+	sgv, err := semver.StrictNewVersion(stencilVersion)
```

Comment:
Was there a reason behind the strict usage? IIRC I've had annoying issues using the strict verison

Reply:
Not particularly, it seemed to be more recommended than `NewVersion`. Happy to use `NewVersion` instead.